test/python/circuit/test_uc.py

```diff
@@ -85,6 +85,22 @@ def test_global_phase_ucg(self):
 
         self.assertTrue(np.allclose(unitary_desired, unitary))
 
+    def test_inverse_ucg(self):
+        """ "Test inverse function of uniformly controlled gates"""
+        gates = [random_unitary(2).data for _ in range(2**2)]
+        num_con = int(np.log2(len(gates)))
+        q = QuantumRegister(num_con + 1)
+        qc = QuantumCircuit(q)
+
+        qc.uc(gates, q[1:], q[0], up_to_diagonal=False)
+        qc.append(qc.inverse(), qc.qubits)
+
+        simulator = BasicAer.get_backend("unitary_simulator")
+        result = execute(qc, simulator).result()
+        unitary = result.get_unitary(qc)
+        unitary_desired = np.identity(2**qc.num_qubits)
+
+        self.assertTrue(np.allclose(unitary_desired, unitary))
```

Comment:
We can just use the `qiskit.quantum_info.Operator` for this, that should be faster for small numbers of qubits 🙂 
```suggestion
        unitary = Operator(qc).data
        unitary_desired = np.identity(2**qc.num_qubits)

        self.assertTrue(np.allclose(unitary_desired, unitary))
```